test/MarketCollateralPool.js

```diff
@@ -21,95 +21,237 @@ contract('MarketCollateralPool', function(accounts) {
+      } catch (err) {
+        settleAndCloseError = err;
+      }
+      assert.ok(settleAndCloseError instanceof Error, 'settleAndClose() did not fail before settlement');
+    });
 
-  });
+    it('should redeem single short tokens after settlement', async function() {
+      // 1. approve collateral and mint tokens
+      const amountToApprove = 1e22;
+      await collateralToken.approve(collateralPool.address, amountToApprove);
+      const qtyToMint = 1;
+      await collateralPool.mintPositionTokens(marketContract.address, qtyToMint, { from: accounts[0]});
+  
+      // 2. transfer part of the long token
+      await longPositionToken.transfer(accounts[1], 1, { from: accounts[0]})
 
-  it(`should redeem single tokens after settlement`, async function() {
+      // 3. force contract to settlement
+      await settleContract();
+  
+      // 4. redeem all short position tokens after settlement should pass
+      const balanceBeforeRedeem = await shortPositionToken.balanceOf.call(accounts[0]);
+      const qtyToRedeem = -1;
+      let error = null;
+      try {
+        await collateralPool.settleAndClose(marketContract.address, qtyToRedeem, { from: accounts[0] });
+      } catch (err) {
+        error = err;
+      } 
+      assert.isNull(error, 'should be able to redeem single tokens after settlement');
 
-  });
+      // 5. balance of short tokens should be updated.
+      const expectedBalanceAfterRedeem = balanceBeforeRedeem.plus(qtyToRedeem);
+      const actualBalanceAfterRedeem = await shortPositionToken.balanceOf.call(accounts[0]);
+      assert.equal(expectedBalanceAfterRedeem.toNumber(), actualBalanceAfterRedeem.toNumber(), 'short position tokens balance was not reduced');
+    });
 
-  it(`should return correct amount of collateral when redeemed after settlement`, async function() {
-
-  });
+    it('should return correct amount of collateral when redeemed after settlement', async function() {
+      // 1. approve collateral and mint tokens
+      const amountToApprove = 1e22;
+      await collateralToken.approve(collateralPool.address, amountToApprove);
+      const qtyToMint = 1;
+      await collateralPool.mintPositionTokens(marketContract.address, qtyToMint, { from: accounts[0]});
+  
+      // 2. transfer part of the long token
+      await longPositionToken.transfer(accounts[1], 1, { from: accounts[0]})
 
-  it('should fail if settleAndClose() is called before settlement', async () => {
-    // let error = null;
-    // let settleAndCloseError = null;
-    // try {
-    //   await collateralPool.settleAndClose.call(marketContract.address, { from: accounts[0] });
-    // } catch (err) {
-    //   settleAndCloseError = err;
-    // }
-    // assert.ok(settleAndCloseError instanceof Error, 'settleAndClose() did not fail before settlement');
+      // 3. force contract to settlement
+      const settlementPrice = await settleContract();
+  
+      // 4. redeem all shorts on settlement
```

Comment:
@perfectmak should we do the same thing here for the long tokens to ensure it works for both sides (long, short) sides of the market.